libs/application-generic/src/usecases/workflow/get-workflow-with-preferences/get-workflow-with-preferences.usecase.ts

```diff
@@ -0,0 +1,67 @@
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
+
+import { NotificationTemplateEntity } from '@novu/dal';
+import { buildWorkflowPreferencesFromPreferenceChannels, DEFAULT_WORKFLOW_PREFERENCES } from '@novu/shared';
+import { GetPreferences, GetPreferencesCommand } from '../../get-preferences';
+
+import { GetWorkflowWithPreferencesCommand } from './get-workflow-with-preferences.command';
+import { WorkflowWithPreferencesResponseDto } from './get-workflow-with-preferences.dto';
+import { Instrument, InstrumentUsecase } from '../../../instrumentation';
+import { GetWorkflowByIdsUseCase } from '../get-workflow-by-ids/get-workflow-by-ids.usecase';
+import { GetWorkflowByIdsCommand } from '../get-workflow-by-ids/get-workflow-by-ids.command';
+
+@Injectable()
+export class GetWorkflowWithPreferencesUseCase {
+  constructor(
+    private getWorkflowByIdsUseCase: GetWorkflowByIdsUseCase,
+    @Inject(forwardRef(() => GetPreferences))
+    private getPreferences: GetPreferences
+  ) {}
+
+  @InstrumentUsecase()
+  async execute(command: GetWorkflowWithPreferencesCommand): Promise<WorkflowWithPreferencesResponseDto> {
+    const workflowEntity = await this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        workflowIdOrInternalId: command.workflowIdOrInternalId,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+      })
+    );
```

Comment:
this is the new part of the code the rest was copied from the old class `GetWorkflowByIdsUseCase`